apstools/utils.py

```diff
@@ -1451,3 +1459,72 @@ def redefine_motor_position(motor, new_position):
     yield from bps.mv(motor.set_use_switch, 1)
     yield from bps.mv(motor.user_setpoint, new_position)
     yield from bps.mv(motor.set_use_switch, 0)
+
+
+def quantify_md_key_use(key=None, db=None, catalog_name=None, since=None, until=None, query={}):
+    """
+    print table of different ``key`` values and how many times each appears
+
+    PARAMETERS
+
+    key : str
+        one of the metadata keys in a run's start document
+        (default: ``plan_name``)
+    db : object
+        Instance of databroker v1 ``Broker`` or v2 ``catalog``
+        (default: see ``catalog_name`` keyword argument)
+    catalog_name : str
+        Name of databroker v2 catalog, used when supplied ``db`` is ``None``.
+        (default: ``mongodb_config``)
+    since : str
+        include runs that started on or after this ISO8601 time
+        (default: ``1995-01-01``)
+    until : str
+        include runs that started before this ISO8601 time
+        (default: ``2100-12-31``)
+    query : dict
+        mongo query dictionary, used to filter the results
+        (default: ``{}``)
+
+        see: https://docs.mongodb.com/manual/reference/operator/query/
+
+    EXAMPLES
+
+        quantify_md_key_use(key="proposal_id")
+        quantify_md_key_use(key="plan_name", catalog_name="9idc", since="2020-07")
+        quantify_md_key_use(key="beamline_id", catalog_name="9idc")
+        quantify_md_key_use(key="beamline_id", 
+                            catalog_name="9idc", 
+                            query={'plan_name': 'Flyscan'}, 
+                            since="2020",
+                            until="2020-06-21 21:51")
+        quantify_md_key_use(catalog_name="8id", since="2020-01", until="2020-03")
+    """
+    key = key or 'plan_name'
+    catalog_name = catalog_name or 'mongodb_config'
+    query = query or {}
+    since = since or "1995-01-01"
+    until = until or "2100-12-31"
+
+    cat = (db or databroker.catalog[catalog_name]).v2.search(
+        databroker.queries.TimeRange(since=since, until=until)
+    ).search(query)
+
+    ids = []
+    while True:
+        runs = cat.search({key:{'$exists': True, '$nin': ids}})
+        if len(runs) == 0:
+            break
+        else:
+            ids.append(runs.v1[-1].start.get(key))
+
+    def sorter(key):
+        if key is None:
+            key = " None"
+        return str(key)
+
+    table = pyRestTable.Table()
+    table.labels = f"{key} #runs".split()
+    for id in sorted(ids, key=sorter):
```

Comment:
Codacy found an issue: [Redefining built-in 'id'](https://app.codacy.com/gh/BCDA-APS/apstools/pullRequest?prid=5939213)